apps/wistia-videos/src/functions/getVideos.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Project,
+  ProjectReduced,
+  WistiaError,
+  WistiaVideo,
+  WistiaProject,
+} from '../components/helpers/types';
+
+const PROJECTS_PER_PAGE = 100;
+const VIDEOS_PER_PAGE = 500;
+
+const fetchProjectsJSON = async (
+  bearerToken: string,
+  page: number,
+  accumulated: WistiaProject[]
+): Promise<WistiaProject[]> => {
+  let projects = await (
+    await fetch(`https://api.wistia.com/v1/projects.json${page ? '?page=' + page : ''}`, {
+      headers: {
+        Authorization: `Bearer ${bearerToken}`,
+      },
+    })
+  ).json();
+  accumulated = [...accumulated, ...projects];
+  if (projects.error) {
```

Comment:
This clause probably belongs just after the fetch call, since we'd probably want to throw right away after fetching as opposed to adding the accumulated resultss?

Reply:
Good call, this is moved up 👍